sklearn/preprocessing/data.py

```diff
@@ -250,13 +312,68 @@ def fit(self, X, y=None):
             raise ValueError("Minimum of desired feature range must be smaller"
                              " than maximum. Got %s." % str(feature_range))
-        data_min = np.min(X, axis=0)
-        data_range = np.max(X, axis=0) - data_min
-        data_range = _handle_zeros_in_scale(data_range)
-        self.scale_ = (feature_range[1] - feature_range[0]) / data_range
-        self.min_ = feature_range[0] - data_min * self.scale_
-        self.data_range = data_range
-        self.data_min = data_min
+
+        if sparse.issparse(X):
+            raise TypeError("MinMaxScaler does no support sparse input.")
+        else:
+            data_min = np.min(X, axis=0)
+            data_max = np.max(X, axis=0)
+            data_range = data_max - data_min
+            self.scale_ = (feature_range[1] - feature_range[0]) / \
+                _handle_zeros_in_scale(data_range)
+            self.min_ = feature_range[0] - data_min * self.scale_
+            self.data_min = data_min
+            self.data_max = data_max
+            self.data_range = data_range
+            self.n_samples_seen_ = X.shape[0]
+        return self
+
+    def partial_fit(self, X, y=None):
+        """Incremental computation of minimum and maximum for later scaling.
+
+        All of X is processed as a single batch. X cannot be a sparse matrix.
+
+        Parameters
+        ----------
+        X : array-like with shape [n_samples_, n_features]
+            The data used to compute the mean and standard deviation
+            used for later scaling along the features axis.
+            X cannot be sparse.
+
+        y: Passthrough for ``Pipeline`` compatibility.
+        """
+        X = check_array(X, copy=self.copy, ensure_2d=False, warn_on_dtype=True,
+                        estimator=self, dtype=FLOAT_DTYPES)
+        # Only accept 2s input, and interpret it as 1-row matrix
+        X = row_or_2d(X)
+
+        feature_range = self.feature_range
+        if feature_range[0] >= feature_range[1]:
+            raise ValueError("Minimum of desired feature range must be smaller"
+                             " than maximum. Got %s." % str(feature_range))
+
+        # First partial_fit
+        if not hasattr(self, 'n_samples_seen_'):
+            self.fit(X)
+            return self
+
+        # Next steps
+        if sparse.issparse(X):
+            raise TypeError("MinMaxScaler does no support sparse input.")
+        else:
+            data_min, data_max = \
+                _incremental_min_and_max(X, axis=0, last_min=self.data_min,
+                                         last_max=self.data_max)
+
+            data_range = data_max - data_min
+            self.scale_ = (feature_range[1] - feature_range[0]) / \
+                _handle_zeros_in_scale(data_range)
+            self.min_ = feature_range[0] - data_min * self.scale_
+            self.data_min = data_min
+            self.data_max = data_max
+            self.data_range = data_range
```

Comment:
all attributes that are computed from the data in the fit method should be suffixed with `_` (scikit-learn convention).